soda-tests/tests/features/test_soda_data_types.py

```diff
@@ -10,3 +17,137 @@ def test_soda_data_types(data_source_test_helper: DataSourceTestHelper):
         str(soda_data_type) for soda_data_type in SodaDataTypeName if soda_data_type not in dialect_soda_data_type_map
     ]
     assert unmapped_data_types == []
+
+
+def test_all_data_types_explicitly(data_source_test_helper: DataSourceTestHelper):
+    mappings_from_canonical_to_data_type: dict = (
+        data_source_test_helper.data_source_impl.sql_dialect.get_data_source_data_type_name_by_soda_data_type_names()
+    )
+
+    # Make sure that each SodaDataType is mapped to a data type in the data source
+    for soda_data_type in SodaDataTypeName:
+        assert soda_data_type in mappings_from_canonical_to_data_type
```

Comment:
Just to make life easier, I suggest adding add a message that indices which data type is missing.

```suggestion
        assert soda_data_type in mappings_from_canonical_to_data_type, f"{soda_data_type} does not have a mapping"
```